apps/modernization-ui/src/apps/page-builder/page/management/edit/staticelement/AddStaticElement.tsx

```diff
@@ -0,0 +1,204 @@
+import { Form, ModalRef, Button, ModalToggleButton } from '@trussworks/react-uswds';
+import {
+    AddDefault,
+    AddHyperlink,
+    AddReadOnlyComments,
+    PageStaticControllerService
+} from 'apps/page-builder/generated';
+import { Input } from 'components/FormInputs/Input';
+import { SelectInput } from 'components/FormInputs/SelectInput';
+import { RefObject } from 'react';
+import { Controller, FormProvider, useForm, useWatch } from 'react-hook-form';
+import { maxLengthRule } from 'validation/entry/maxLengthRule';
+import { HyperlinkFields } from './HyperlinkFields';
+import { CommentsFields } from './CommentsFields';
+import { authorization } from 'authorization';
+import { usePageManagement } from '../../usePageManagement';
+import styles from './staticelement.module.scss';
+import { useAlert } from 'alert';
+
+const staticType = [
+    { value: 'LIN', name: 'Line separator' },
+    { value: 'HYP', name: 'Hyperlink' },
+    { value: 'COM', name: 'Comments (read-only)' },
+    { value: 'PAR', name: 'Participant list (read-only)' },
+    { value: 'ELE', name: 'Electronic document list (read-only)' }
+];
+
+type AddStaticElementModalProps = {
+    modalRef?: RefObject<ModalRef>;
+    subsectionId?: number;
+};
+
+type StaticElementType = {
+    type: 'HYP' | 'COM' | 'LIN' | 'PAR' | 'ELE' | '';
+};
+
+type StaticElementFormValues = (AddReadOnlyComments | AddHyperlink | AddDefault) & StaticElementType;
+
+export const AddStaticElement = ({ modalRef, subsectionId }: AddStaticElementModalProps) => {
+    const form = useForm<StaticElementFormValues>({
+        mode: 'onBlur'
+    });
+    const watch = useWatch({ control: form.control });
+    const { page } = usePageManagement();
+    const { showAlert } = useAlert();
+
+    const handleSubmit = () => {
+        onSubmit();
+    };
+
+    const handleAlert = (message: string) => {
+        showAlert({ message: message, type: 'success' });
+    };
+
+    const onSubmit = form.handleSubmit(async (data) => {
+        switch (data.type) {
+            case 'HYP': {
+                data.subSectionId = subsectionId;
+                PageStaticControllerService.addStaticHyperLinkUsingPost({
+                    authorization: authorization(),
+                    page: page.id,
+                    request: data
+                });
+                form.reset();
```

Comment:
issue (blocking):  The form should be reset and the alert shown once we know that the API is finished.  This can be done by wrapping those two calls in the `then` of the promise provided by the API call.

```
      PageStaticControllerService.addStaticHyperLinkUsingPost({
                    authorization: authorization(),
                    page: page.id,
                    request: data
                }).then(() => {
                    form.reset();
                    handleAlert(`The element ${(data as AddHyperlink).label} has been successfully added.`);
                });
```